docs/running-on-yarn.md

```diff
@@ -134,6 +134,13 @@ Most of the configs are the same for Spark on YARN as for other deployment modes
   </td>
 </tr>
 <tr>
+  <td><code>spark.yarn.am.port</code></td>
+  <td>(random)</td>
+  <td>
+    Port for the YARN Application Master to listen on. In yarn client mode, this is used to communicate between the spark driver running on a gateway and the application master running on yarn. In yarn cluster mode, this is used for the dynamic executor feature, where it handles the kill from the scheduler backend.
```

Comment:
Nits: yarn -> YARN, spark -> Spark, 'application master' -> 'Application Master'. Is this really the most accurate description of the AM role? it's where you find the Spark UI, right? the rest of this info isn't so useful to callers. It's just the port where the AM listens.